extensions/amp-iframe/0.1/amp-iframe.js

```diff
@@ -177,104 +196,153 @@ export class AmpIframe extends AMP.BaseElement {
+              'https://github.com/ampproject/amphtml/issues/new');
+          return Promise.resolve();
+        }
+        console/*OK*/.error('It looks like this page contains an iframe that' +
+            ' is used for tracking or analytics purposes. Please use ' +
+            'amp-analytics instead. This usage of amp-iframe will break ' +
+            'in the future');
+      }
 
-    /** @private @const {!HTMLIFrameElement} */
-    this.iframe_ = iframe;
+      const width = this.element.getAttribute('width');
+      const height = this.element.getAttribute('height');
+      const iframe = document.createElement('iframe');
 
-    this.applyFillContent(iframe);
-    iframe.width = getLengthNumeral(width);
-    iframe.height = getLengthNumeral(height);
-    iframe.name = 'amp_iframe' + count++;
+      this.applyFillContent(iframe);
+      iframe.width = getLengthNumeral(width);
+      iframe.height = getLengthNumeral(height);
+      iframe.name = 'amp_iframe' + count++;
 
-    if (this.isClickToPlay_) {
-      iframe.style.zIndex = -1;
-    }
+      if (this.isClickToPlay_) {
+        iframe.style.zIndex = -1;
+      }
 
-    /** @private @const {boolean} */
-    this.isResizable_ = this.element.hasAttribute('resizable');
-    if (this.isResizable_) {
-      this.element.setAttribute('scrolling', 'no');
-      assert(this.getOverflowElement(),
-          'Overflow element must be defined for resizable frames: %s',
-          this.element);
-    }
+      if (this.isResizable_) {
+        assert(this.getOverflowElement(),
+               'Overflow element must be defined for resizable frames: %s',
+               this.element);
+      }
 
-    /** @const {!Element} */
-    this.propagateAttributes(
-        ['frameborder', 'allowfullscreen', 'allowtransparency', 'scrolling'],
-        iframe);
-    setSandbox(this.element, iframe, this.sandbox_);
-    iframe.src = this.iframeSrc;
-    this.element.appendChild(makeIOsScrollable(this.element, iframe));
-    if (!isTracking) {
-      /** @private {!IntersectionObserver} */
-      this.intersectionObserver_ =
-          new IntersectionObserver(this, this.iframe_);
-    }
+      /** @const {!Element} */
+      this.propagateAttributes(
+          ['frameborder', 'allowfullscreen', 'allowtransparency', 'scrolling'],
+          iframe);
+      iframe.src = this.iframeSrc;
 
-    iframe.onload = () => {
-      // Chrome does not reflect the iframe readystate.
-      iframe.readyState = 'complete';
-      this.activateIframe_();
-      if (isTracking) {
-        timer.promise(trackingIframeTimeout).then(() => {
-          if (iframe.parentNode) {
-            iframe.parentNode.removeChild(iframe);
-          }
-          this.element.setAttribute('amp-removed', '');
-          this.iframe_ = null;
-          this.iframeSrc = null;
-        });
+      if (!isTracking) {
+        /** @private {!IntersectionObserver} */
+        this.intersectionObserver_ = new IntersectionObserver(this, iframe);
```

Comment:
Is `IntersectionObserver` created before the iframe is added to DOM? If so, this could be problematic since it needs to access `contentWindow`.